src/modularity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+    randompartitioning(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
+    
+Generates a random `n`-partitioning for a hypergraph `hg`. 
+"""
+function randompartitioning(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
+    x = rand(1:n, size(hg, 1))
+    [findall(==(i), x) for i in 1:n]
+end
+
+"""
+    modularity(hg::Hypergraph, partitioning::Vector{Vector{Int}})
+    
+Calculates the modularity of a hypergraph `hg` for a given `partitioning`.
+"""
+function modularity(hg::Hypergraph, partitioning::Vector{Vector{Int}})
+    @assert sum(length.(partitioning)) == size(hg, 1)
+    @assert sort!(union(partitioning...)) == axes(hg, 1)
+    hes = [count(==(true), hg[:, i]) for i in axes(hg, 2)]
+    Ed = Dict{Int,Int}()
+    for i in 2:maximum(hes)
```

Comment:
it is possible to populate `Ed[i]` using one pass over `hes`

Reply:
rewrote the code - with the test example 20x performance increase is observed.